core/src/main/java/de/danoeh/antennapod/core/service/PlayerWidgetJobService.java

```diff
@@ -91,6 +96,22 @@ private void updateViews() {
         if (media != null) {
             views.setOnClickPendingIntent(R.id.layout_left, startMediaplayer);
 
+            try {
+                Bitmap icon = null;
+                int iconSize = getResources().getDimensionPixelSize(
+                        android.R.dimen.app_icon_size);
+                icon = Glide.with(PlayerWidgetJobService.this)
```

Comment:
I believe you could easily move this off the main thread with RxJava

Reply:
No matter what else is pushed to the background, the main UI thread still has to update the ImageView, that might make widget unresponsive.